crates/ide_assists/src/handlers/generate_function.rs

```diff
@@ -44,13 +44,16 @@ use crate::{
 pub(crate) fn generate_function(acc: &mut Assists, ctx: &AssistContext) -> Option<()> {
     let path_expr: ast::PathExpr = ctx.find_node_at_offset()?;
     let call = path_expr.syntax().parent().and_then(ast::CallExpr::cast)?;
-    let path = path_expr.path()?;
 
+    let path = path_expr.path()?;
     if ctx.sema.resolve_path(&path).is_some() {
         // The function call already resolves, no need to add a function
         return None;
     }
 
+    let await_expr = call.syntax().parent().and_then(ast::AwaitExpr::cast);
+    let is_async = await_expr.is_some();
```

Comment:
You can leave this inside the `FunctionBuilder::from_call` since we still have access to `call` in there, no need to pass the `is_async` as a parameter then.

Looks good to me otherwise
bors d+